lib/puppet/pops.rb

```diff
@@ -10,10 +10,13 @@ module Puppet
   #
   # @api public
   module Pops
-     EMPTY_HASH = {}.freeze
+    EMPTY_HASH = {}.freeze
     EMPTY_ARRAY = [].freeze
     EMPTY_STRING = ''.freeze
 
+    MAX_INTEGER =  0x7fffffffffffffff
+    MIN_INTEGER = -0x8000000000000000
+
```

Comment:
Not sure if we want 59640dd6d (https://tickets.puppetlabs.com/browse/PUP-3170) in 4.10.x /cc @hlindberg @thallgren It adds a check against overflow/underflow.